src/main/kotlin/com/avail/interpreter/levelTwo/operation/L2_PHI_PSEUDO_OPERATION.kt

```diff
@@ -0,0 +1,360 @@
+		while (iterator.hasNext())
+		{
+			restriction = restriction.union(iterator.next().restriction())
+		}
+		registerSet.removeConstantAt(destination.register())
+		registerSet.removeTypeAt(destination.register())
+		registerSet.typeAtPut(
+			destination.register(), restriction.type, instruction)
+		restriction.constantOrNull?.let {
+			registerSet.constantAtPut(
+				destination.register(),
+				it,
+				instruction)
+		}
+	}
+
+	override val isPhi: Boolean
+		get() = true
+
+	override fun instructionWasAdded(
+		instruction: L2Instruction, manifest: L2ValueManifest)
+	{
+		// The reads in the input vector are from the positionally corresponding
+		// incoming edges, which carry the manifests that should be used to
+		// look up the best source semantic values.
+		val sources: L2ReadVectorOperand<RR, R> = instruction.operand(0)
+		val destination: L2WriteOperand<R> = instruction.operand(1)
+		val predecessorEdges =
+			instruction.basicBlock().predecessorEdgesCopy()
+		instructionWasAddedForPhi(sources, predecessorEdges)
+		destination.instructionWasAdded(manifest)
+	}
+
+	override fun shouldEmit(
+		instruction: L2Instruction): Boolean
+	{
+		// Phi instructions are converted to moves along predecessor edges.
+		return false
+	}
+
+	/**
+	 * One of this phi function's predecessors has been removed because it's
+	 * dead code.  Clean up its vector of inputs by removing the specified
+	 * index.
+	 *
+	 * @param instruction
+	 *   The [L2Instruction] whose operation has this type.
+	 * @param inputIndex
+	 *   The index to remove.
+	 * @return
+	 *   A replacement [L2Instruction], whose operation may be either another
+	 *   `L2_PHI_PSEUDO_OPERATION` or an [L2_MOVE].
+	 */
+	fun withoutIndex(
+		instruction: L2Instruction,
+		inputIndex: Int): L2Instruction
+	{
+		val oldVector: L2ReadVectorOperand<RR, R> = instruction.operand(0)
+		val destinationReg: L2WriteOperand<R> = instruction.operand(1)
+		val newSources: MutableList<RR> = ArrayList(oldVector.elements())
+		newSources.removeAt(inputIndex)
+		val onlyOneRegister = HashSet(newSources).size == 1
+		return if (onlyOneRegister)
+		{
+			// Replace the phi function with a simple move.
+			L2Instruction(
+				instruction.basicBlock(),
+				moveOperation,
+				newSources[0],
+				destinationReg)
+		}
+		else L2Instruction(
+			instruction.basicBlock(),
+			this,
+			oldVector.clone(newSources),
+			destinationReg)
+	}
+
+	/**
+	 * Replace this phi by providing a lambda that alters a copy of the list of
+	 * [L2ReadOperand]s that it's passed.  The predecessor edges are expected to
+	 * correspond with the inputs.  Do not attempt to normalize the phi to a
+	 * move.
+	 *
+	 * @param instruction
+	 *   The phi instruction to augment.
+	 * @param updater
+	 *   What to do to a copied mutable [List] of read operands that starts out
+	 *   having all of the vector operand's elements.
+	 */
+	fun updateVectorOperand(
+		instruction: L2Instruction,
+		updater: Consumer<MutableList<RR>>)
```

Comment:
Use Kotlin function type:
`... updater: (MutableList<RR>)->Unit)`
Also, I believe this function can be made private.